File: Atcoder_#350/javascript/b_dentist_aoki.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question('', (line1) => {
  rl.question('', (line2) => {
    const [n, q] = line1.split(' ').map(Number);
    const t = line2.split(' ').map(Number);

    // 穴ごとの治療回数をカウントするオブジェクトを作成
    const treatmentCounts = {};

    // 各治療回数をカウント
    t.forEach(i => {
      treatmentCounts[i] = (treatmentCounts[i] || 0) + 1;
    });

    // 奇数回の治療が行われた穴の数を数える
    let oddTreatmentCount = 0;
    for (const count of Object.values(treatmentCounts)) {
      if (count % 2 !== 0) {
        oddTreatmentCount += 1;
      }
    }

    // 最終的に残っている歯の数を計算
    const remainingTeeth = n - oddTreatmentCount;

    // 結果を出力
    console.log(remainingTeeth);

    rl.close();
  });
});

File: Atcoder_#350/ruby/b_dentist_aoki.rb
n, q = gets.chomp.split.map(&:to_i)
t = gets.chomp.split.map(&:to_i)

# 穴ごとの治療回数をカウントするハッシュを作成
treatment_counts = Hash.new(0)

# 各治療回数をカウント
t.each do |i|
  treatment_counts[i] += 1
end

# 奇数回の治療が行われた穴の数を数える
odd_treatment_count = treatment_counts.values.count { |count| count.odd? }

# 最終的に残っている歯の数を計算
remaining_teeth = n - odd_treatment_count

# 結果を出力
puts remaining_teeth
